pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -454,7 +456,14 @@ def register_plugins(cls, plugin_registry: Dict) -> None:
             find_unused_parameters=False,
         )
 
-    def _share_information_to_prevent_deadlock(self):
+    def _share_information_to_prevent_deadlock(self) -> None:
+        self._has_called_call_children_scripts = self.broadcast(self._has_called_call_children_scripts)
+
+        # Short-circuit debug info set for process reconciliation if processes
+        # are managed by an scheduler or parent process external to Lightning
+        if not self._has_called_call_children_scripts:
+            return
+
```

Comment:
note to reviewer: this is a perf optimization

for slurm/torchelastic, there's no need to run the all_gather/broadcasts after this if we're not going to use them in `reconciliate_processes`